ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/OnChainPrivacyPrecompiledContract.java

```diff
@@ -226,11 +242,97 @@ boolean canExecute(
         blockchain,
         disposablePrivateState,
         privateWorldStateUpdater)) {
+      LOG.debug(
+          "Privacy group version mismatch while trying to execute transaction with commitment {}",
+          messageFrame.getTransactionHash());
+      return false;
+    }
+
+    if (!isMemberOfPrivacyGroup(
+        isAddingParticipant,
+        privateTransaction,
+        privateFrom,
+        privacyGroupId,
+        messageFrame,
+        currentBlockHeader,
+        publicWorldState,
+        blockchain,
+        disposablePrivateState,
+        privateWorldStateUpdater)) {
+      LOG.debug(
+          "PrivateTransaction with hash {} cannot execute in privacy group {} because private from {} is not a member.",
```

Comment:
should this say "privateFrom" rather than "private from"?

Reply:
Done